### DISCUSSION TITLE
Why useFlexLayout set width

### DISCUSSION THREAD

**Discussion by matthewma7** [TanStack/table#2022] [General] [answered]
I understand that useFlexLayout would set `flex`, but why does it [also set the width](https://github.com/tannerlinsley/react-table/blob/master/src/plugin-hooks/useFlexLayout.js#L40)?
Shouldn't width be totally controlled by flex?

**Comment by matthewma7** [comment, marked as answer]
I feel it may have something to do with resizable columns. Without the need of resizable. 
This simplified version of useFlexLayout worked better for me in terms of layout and width sizing. 

[useMyFlexLayout.js.txt](https://github.com/tannerlinsley/react-table/files/4355326/useMyFlexLayout.js.txt)

**Reply by dbousamra** [reply]
Thanks. This works perfectly. 

Here's the slightly modified code I used:

```typescript
export default function useMyFlexLayout(hooks: any) {
  const getRowStyles = (props: any, { instance }: any) => [
    props,
    {
      style: {
        display: 'flex',
      },
    },
  ];

  const getHeaderProps = (props: any, { column }: any) => [
    props,
    {
      style: {
        flex: column.width,
        minWidth: column.minWidth,
      },
    },
  ];

  const getCellProps = (props: any, { cell }: any) => {
    return [
      props,
      {
        style: {
          flex: cell.column.width,
          minWidth: cell.column.minWidth,
        },
      },
    ];
  };

  hooks.getRowProps.push(getRowStyles);
  hooks.getHeaderGroupProps.push(getRowStyles);
  hooks.getHeaderProps.push(getHeaderProps);
  hooks.getCellProps.push(getCellProps);
}
```

**Reply by crbanman** [reply]
These two solutions are exactly what I needed. I ran into an issue where if I wanted to use virtualized rows I had to give up the ability for the table to be a dynamic width, even with `useFlexLayout`. I tried playing around with `react-virtualized-auto-sizer`, but that did more bad than good.

For anyone running into the same problem, use the [virtualized tables example](https://react-table.tanstack.com/docs/examples/virtualized-rows) but replace the use of `useBlockLayout` with this plugin and don't set a width on the `FixedSizeList`.

**Reply by ecnaidar** [reply]
Thanks for your solutions!
Quite weird that there is no way to disable `width` provision in standard plugin.
